db-tests/src/test/java/gov/nasa/jpl/aerie/database/ActivityDirectiveChangelogTests.java

```diff
@@ -0,0 +1,291 @@
+          // language=sql
+          """
+          UPDATE activity_directive
+          SET start_offset = '%s'
+          WHERE plan_id = %d and id = %d
+          RETURNING id;
+          """.formatted("01:01:01", planId, activityId));
+      updatedRows.next();
+      assertEquals(activityId, updatedRows.getInt(1));
+    }
+
+    assertEquals(2, getChangelogRevisionCount(planId, activityId));
+}
+    @Test
+    void changelogRevisionHasCorrectValues() throws SQLException {
+      final var activityId = merlinHelper.insertActivity(planId);
+
+      try (final var statement = connection.createStatement()) {
+        final var res = statement.executeQuery(
+            // language=sql
+            """
+            SELECT *
+            FROM activity_directive_changelog
+            WHERE plan_id = %d and activity_directive_id = %d;
+            """.formatted(planId, activityId));
+
+        res.next();
+        final var current = getActivity(planId, activityId);
+
+        assertEquals(current.activityId(), res.getInt("activity_directive_id"));
+        assertEquals(current.planId(), res.getInt("plan_id"));
+        assertEquals(current.name(), res.getString("name"));
+        assertEquals(current.sourceSchedulingGoalId(), res.getInt("source_scheduling_goal_id"));
+        assertEquals(current.lastModifiedAt(), res.getString("changed_at"));
+        assertEquals(current.lastModifiedBy(), res.getString("changed_by"));
+        assertEquals(current.startOffset(), res.getString("start_offset"));
+        assertEquals(current.type(), res.getString("type"));
+        assertEquals(current.arguments(), res.getString("arguments"));
+        assertEquals(current.lastModifiedArgumentsAt(), res.getString("changed_arguments_at"));
+        assertEquals(current.metadata(), res.getString("metadata"));
+        assertEquals(current.anchorId(), res.getString("anchor_id"));
+        assertEquals(current.anchoredToStart(), res.getBoolean("anchored_to_start"));
+    }
+  }
+
+  @Test
+  void shouldDeleteChangelogsOverRevisionLimit() throws SQLException {
+    final var maxRevisionsLimit = 11;
+    final var activityId = merlinHelper.insertActivity(planId);
+
+    // randomly update activity directive > maxRevisionsLimit times
+    for (int i = 0; i < maxRevisionsLimit * 2; i++) {
+      connection.createStatement()
+                .executeQuery(
+                    // language=sql
+                    """
+                    UPDATE activity_directive
+                    SET start_offset = '%02d'
+                    WHERE plan_id = %d and id = %d
+                    RETURNING id;
+                    """.formatted(i, planId, activityId)
+                )
+                .close();
+    }
+
+    assertEquals(maxRevisionsLimit, getChangelogRevisionCount(planId, activityId));
+  }
+
+  @Test
+  void revertNonExistentPlanThrowsError() throws SQLException {
+    try {
+      revertActivityDirectiveToChangelog(-1, -1, -1);
```

Comment:
Why was the choice made to have all the parameters of the function be invalid rather than just the one under examination (`planId` in this case)?

Reply:
To check that the plan exception is thrown first, even if the other params are invalid as well.